enumerables.rb

```diff
@@ -0,0 +1,129 @@
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |item|
+      new_array << item if yield(item) == true
+    end
+    new_array.empty? ? self : new_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return false unless yield(item)
+      elsif pattern.nil?
+        return false unless item
+      else
+        return false unless pattern === item
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return true unless yield(item) == false
+      elsif pattern.nil?
+        return true unless item
+      else
+        return true unless (pattern === item) == false
+      end
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    new_array = []
+    my_each do |item|
+      new_array << if block_given?
+                     (yield(item) == true ? 1 : 0)
+                   elsif pattern.nil?
+                     (item == true ? 1 : 0)
+                   else
+                     ((pattern === item) == true ? 1 : 0)
+                   end
+    end
+    count = 0
+    new_array.my_each do |x|
+      count += x
+    end
+    count.zero?
+  end
+
+  def my_count(num = 0, &block)
+    result = 0
+    my_each do |item|
+      if block_given? == false and item == num
+        result += 1
+      elsif block_given? and block.call(item) == true
+        result += 1
+      end
+    end
+    result != 0 ? result : my_each { |x| x }
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/81323162-2c1c6d80-9052-11ea-8f65-9fd0dc71be81.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)